jest.config.ts

```diff
@@ -0,0 +1,88 @@
+import type {Config} from "jest";
+import os from "node:os";
+import pathPosix from "node:path/posix";
+
+
+let PRIMARY_REPORTER: string | [string, Record<string, unknown>] = "default";
+if ("undefined" !== typeof process.env.CI) {
+    PRIMARY_REPORTER = [
+        "github-actions",
+        {silent: false},
+    ];
+}
+console.log(`Environment variable "CI"="${process.env.CI}": ` +
+    `primary reporter will be ${JSON.stringify(PRIMARY_REPORTER)}`);
+
+const JEST_CONFIG: Config = {
+    collectCoverage: true,
+    collectCoverageFrom: ["src/**/*.{ts,tsx}"],
+    coverageProvider: "v8",
+    coverageReporters: ["text"],
+    coverageThreshold: {
+        "global": {
+            functions: 90,
+            lines: 90,
+        },
+        // eslint-disable-next-line no-warning-comments
+        // TODO: Remove / Adjust below overrides as more test cases are added.
+        "src/*.tsx": {
+            functions: 0,
+            lines: 0,
+        },
+        "src/components/": {
+            functions: 0,
+            lines: 0,
+        },
+        "src/contexts/": {
+            functions: 0,
+            lines: 0,
+        },
+        "src/services/": {
+            functions: 0,
+            lines: 0,
+        },
+        "src/typings/": {
+            functions: 0,
+            lines: 0,
+        },
+        "src/utils/": {
+            functions: 0,
+            lines: 0,
+        },
+        "src/utils/math.ts": {
+            functions: 100,
+            lines: 100,
+        },
+    },
+    displayName: {
+        name: "yscope-log-viewer",
+        color: "blue",
+    },
+    errorOnDeprecated: true,
+    maxConcurrency: os.cpus().length,
+    maxWorkers: "100%",
+    openHandlesTimeout: 1000,
+    reporters: [
+        PRIMARY_REPORTER,
+        "summary",
+    ],
+    showSeed: true,
+    testEnvironment: "node",
+    testMatch: [
+        pathPosix.join(__dirname, "test/**/?(*)test.{ts,tsx}"),
+    ],
+    testTimeout: 5000,
+    transform: {
+        ".(ts|tsx)$": [
+            "ts-jest",
+            {useESM: true},
+        ],
+    },
+    verbose: true,
+
+    // Caution: Extra properties set on `Error`, `Map` or `Set` will not be passed on through the
+    // serialization.
```

Comment:
```suggestion
    // Caution: Extra properties set on `Error`, `Map`, or `Set` objects in messages will not be
    // serialized.
```

Reply:
Is this for messages returned from a test?

Reply:
If using `process` (instead of `thread`) workers, there are cases the "received" (actual) result cannot be serialized / deserialized because they're unsupported by `JSON.strigify()`. See https://github.com/klesun-productions/jest-bigint-worker-issue
With that said, I haven't been able to craft an example that confirms the missing "extra properties" with `workerThreads: true`.

Reply:
Gotcha, so I guess my suggestion is more or less accurate?

Reply:
yeah, it's not about the messages. How about
```suggestion
    // Caution: Extra properties set on received values of type `Error`, `Map`, or `Set` will not
    // be passed on through the serialization.
    // See https://jestjs.io/docs/29.6/configuration#workerthreads
```

Reply:
* Received from what?
* Serialization to where?

Reply:
* Received from the process worker runners
* Serialization to the test main process / thread.
(if I didn't understand the arch wrong

Reply:
Sorry, I mean what's sending a value from the worker? I get that the worker is sending something to the main thread, just not sure where that value is coming from.

Reply:
According to their source code, I think your original assumption about the serialization in messages is right. Let me further confirm that.

Reply:
Here's how the test run can be accelerated by multiple jest workers:
1. Their workers are implemented with either (Worker) threads or processes. 
2. The parent process spawns children workers to run individual test scripts (e.g. `module1_test.js`, `module2_test.js`).
3. The children, after finishing the test and composing a message (which already includes the stringified `expected`/`received` values so the issue isn't in the message), will send back a result object which includes both the message and a `matcherResult` field which includes the raw `expected` / `received` values: https://github.com/jestjs/jest/blob/22029ba06b69716699254bb9397f2b3bc7b3cf3b/packages/expect/src/index.ts#L333
4. The result object is sent back to the parent process via channels provided by Node.js. In the process implementation, JSON serialization of the result object happens when `process.send(result)` is called; on the other hand, in the thread implementation the result object is simply deeply cloned (haven't fully confirmed this yet) before returning to the parent. 
5. JS JSON does not have natively support for BigInt, which was why the issue was observed in the above repo; the thread implementation uses `structuredClone` to copy the object, so there was no issue raised with the BigInt if with thread runners.

With those said, I don't really think enabling `workerThreads: true` would impact us, because the test report formatters we are using don't reference the `matcherResult` field of the result object.

For the comment, how about:
```suggestion
    // Note: Extra properties in types such as `Error`, `Map`, or `Set` are not  preserved when
    // passing payloads between parent and child threads due to serialization.
    // See https://jestjs.io/docs/29.6/configuration#workerthreads
```

Reply:
Thanks for the explanation.

Slight edit to the suggestion:

```suggestion
    // NOTE: Extra properties in types such as `Error`, `Map`, or `Set` are not preserved when
    // passing payloads between parent and child threads due to serialization.
    // See https://jestjs.io/docs/29.6/configuration#workerthreads
```

(You'll note that I use NOTE for notes, to make them notably notable.)